File: java/program/Mahasiswa.java


public class Mahasiswa extends Human {
    
    /*
     * Kelas yang berisi data2 mahasiswa.
     * Terdiri ddari atribut name, nim, prodi, fakultas
     * Masing2 atribut dapat diset dan diambil menggunakan methode
     */
    private String nim;
    private String prodi;
    private String fakultas;
    protected Human m_human;

    public Mahasiswa(String name, String nim, String prodi, String fakultas, String nik, GENDER gender) {
        super(name, nik, gender);
        this.name = name;
        this.nim = nim;
        this.prodi = prodi;
        this.fakultas = fakultas;
    }

    // Setter
    public void setNim(String nim)
    {
        this.nim = nim;
    }

    public void setProdi(String prodi) {
        this.prodi = prodi;
    }

    public void setFakultas(String fakultas) {
        this.fakultas = fakultas;
    }

    // Getter
    public String getNim()
    {
        return this.nim;
    }

    public String getProdi()
    {
        return this.prodi;
    }

    public String getFakultas() {
        return this.fakultas;
    }
}

File: python/program/Mahasiswa.py
from Human import *

class Mahasiswa(Human):
    '''
    Kelas untuk menampung data2 Mahasiswa
    terdiri dari Nama, Nim, Prodi, dan Fakultas
    Masing2 dapat diset menggunakan setter
    Masing2 dapat diambil menggunakan getter
    '''
    
    def __init__(self, name, nim, gender, prodi, fakultas, nik):
        self._name = name
        self._nim = nim
        self._gender = gender
        self._prodi = prodi
        self._fakultas = fakultas
        self._nik = nik
    
    #Setter
    
    def setNim(self, nim):
        self._nim = nim
    
    def setProdi(self, prodi):
        self._prodi = prodi
        
    def setFac(self, fakultas):
        self.fakultas = fakultas
    
    #getter
    
    def getNim(self):
        return self._nim
    
    def getProdi(self):
        return self._prodi
    
    def getFac(self):
        return self._fakultas